library/Zend/Controller/Request/Http.php

```diff
@@ -313,6 +315,8 @@ public function setPost($spec, $value = null)
      * @param string $key
      * @param mixed $default Default value to use if key not found
      * @return mixed Returns null if key does not exist
+     *
+     * @phpstan-return ($key is null ? array<string, mixed> : mixed)
```

Comment:
even if these annotations are prefixed with `phpstan-` they also work for `psalm`